processor/tailsamplingprocessor/sampling/string_tag_filter.go

```diff
@@ -65,7 +72,7 @@ func (saf *stringAttributeFilter) Evaluate(_ pdata.TraceID, trace *TraceData) (D
 			rs := rspans.At(i)
 			resource := rs.Resource()
 			if v, ok := resource.Attributes().Get(saf.key); ok {
-				if _, ok := saf.values[v.StringVal()]; ok {
+				if ok := saf.Matches(v.StringVal()); ok {
```

Comment:
Could you add a benchmark to ensure that we are not having performance penalties for the non-regex case? This code is on the critical path, analyzing all attributes of all spans flowing through this collector.

Reply:
Added the benchmark comparison result (before & after). It actually brought some performance regression with supporting regex. Any suggestion?

Reply:
I would keep the regex as a separate policy, leaving the string one untouched.

Reply:
Thx. I sent the updates to extend `StringAttributePolicy` with the new optional config attributes for supporting regex string matching. By default it will apply strict value match without harming the current performance. And users could enable regex match on-demand. 

PTAL! Ty!